src/rendering.cpp

```diff
@@ -111,62 +138,45 @@ bool renderWorldModel(const ed::WorldModel& world_model, const enum ShowVolumes
     geo::Mesh y_box = geo::Box(geo::Vector3(-at, 0, -at), geo::Vector3(at, al, at)).getMesh();
     geo::Mesh z_box = geo::Box(geo::Vector3(-at, -at, 0), geo::Vector3(at, at, al)).getMesh();
 
-    res.setColor(cv::Vec3b(0, 0, 255));
-    res.setMesh(&x_box);
-    opt.setMesh(x_box, cam_pose_inv);
-    cam.render(opt, res);
-
-    res.setColor(cv::Vec3b(0, 255, 0));
-    res.setMesh(&y_box);
-    opt.setMesh(y_box, cam_pose_inv);
-    cam.render(opt, res);
-
-    res.setColor(cv::Vec3b(255, 0, 0));
-    res.setMesh(&z_box);
-    opt.setMesh(z_box, cam_pose_inv);
-    cam.render(opt, res);
+    renderMesh(cam, cam_pose_inv, x_box, cv::Vec3b(0, 0, 255), res, flatten);
+    renderMesh(cam, cam_pose_inv, y_box, cv::Vec3b(0, 255, 0), res, flatten);
+    renderMesh(cam, cam_pose_inv, z_box, cv::Vec3b(255, 0, 0), res, flatten);
 
     for(ed::WorldModel::const_iterator it = world_model.begin(); it != world_model.end(); ++it)
     {
         const ed::EntityConstPtr& e = *it;
         const std::string& id = e->id().str();
 
-        if (e->shape() && e->has_pose() && (id.size() < 5 || id.substr(id.size() - 5) != "floor")) // Filter ground plane
+        if (e->shape() && e->has_pose() && !e->hasFlag("self") && (id.size() < 5 || id.substr(id.size() - 5) != "floor")) // Filter ground plane
```

Comment:
Why do you not want to render the robot itself?

Reply:
This is currently used by the model viewer, then there is no robot. And the new map service. In which I don't like to see the robot. Especially also because HERO is not shown fully. As some meshes are problematic.